src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/UpdateSentenceDateService.kt

```diff
@@ -51,18 +57,18 @@ class UpdateSentenceDateService(
       staffMember = staffMember,
     )
 
-    licenceRepository.saveAndFlush(updatedLicenceEntity)
-
-    auditEventRepository.saveAndFlush(
-      AuditEvent(
-        licenceId = licenceEntity.id,
-        username = "SYSTEM",
-        fullName = "SYSTEM",
-        eventType = AuditEventType.SYSTEM_EVENT,
-        summary = "Sentence dates updated for ${licenceEntity.forename} ${licenceEntity.surname}",
-        detail = "ID ${licenceEntity.id} type ${licenceEntity.typeCode} status ${licenceEntity.statusCode} version ${licenceEntity.version}",
-      ),
-    )
+    val licenceIsNowInHardStopPeriod = releaseDateService.isInHardStopPeriod(updatedLicenceEntity)
+
+    if (licenceIsNowInHardStopPeriod && updatedLicenceEntity is CrdLicence) {
+      val timedOutLicence = updatedLicenceEntity.timeOut()
+      licenceRepository.saveAndFlush(timedOutLicence)
+      updatedLicenceEntity.recordAuditEvent("Sentence dates updated")
+      timedOutLicence.recordAuditEvent("Licence automatically timed out")
+      timedOutLicence.recordLicenceEvent("Licence automatically timed out")
```

Comment:
```
updatedLicenceEntity.recordAuditEvent("Sentence dates updated")
timedOutLicence.recordAuditEvent("Licence automatically timed out")
```

these two licences are the "same" (share the same id) - would just refer to the updated one after it's been updated. (the other one can be seen as a stale copy)

Reply:
Done